src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/_params.py

```diff
@@ -1229,3 +1230,42 @@ def _configure_security_policy_storage_params(arg_ctx):
                    id_part='name')
 
         c.argument('resource_group_name', arg_type=resource_group_name_type)
+
+    ###############################################
+    #             sql instance failover-group     #
+    ###############################################
+
+    with self.argument_context('sql instance-failover-group') as c:
+        c.argument('instance_failover_group_name',
+                   options_list=['--name', '-n'],
+                   help="The name of the Instance Failover Group")
+
+        c.argument('failover_group_name',
+                   options_list=['--name', '-n'],
+                   help="The name of the Instance Failover Group")
+
+        c.argument('managed_instance',
+                   arg_type=managed_instance_param_type,
+                   options_list=['-s'])
+
+        c.argument('partner_managed_instance',
+                   help="The name of the partner managed instance of a Instance Failover Group")
+
+        c.argument('partner_resource_group',
+                   help="The name of the resource group of the partner managed instance")
```

Comment:
If partner managed instance is an existing resource, then this violates command guidelines:
https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md#supporting-name-or-id-parameters

Reply:
if we constructed this parameter called partner_managed_instance and have customers pass in the name of resource id of this managed_instance, does it imply that the server belongs in the current user subscription? Because if we only pass in the name, I assume the ARM resource id will be constructed using the current resource group/subscription id.

This is not possible because we will need to support cross resource group/subscription pairing for this feature. Any thoughts on that?

Reply:
The crossRG/cross-subscription story in CLI is to use ARM ID.